src/renderer/store/actions.js

```diff
@@ -414,12 +414,23 @@ export async function playTrackIndex({ state, commit, dispatch }, index) {
 /**
  * @param {ActionContext} param0
  */
-export function playNextTrack({ dispatch, getters }) {
-    const { index, list, loopMode } = getters.queue;
+export function playNextTrack({ commit, dispatch, getters }) {
+    let { index, list, loopMode, randomHeardList, randomHeardListPointer } = getters.queue;
     let nextIndex;
     switch (loopMode) {
         case LOOP_MODE.RANDOM:
-            nextIndex = Math.floor(Math.random() * list.length);
+            if (randomHeardList.length == 0) {
+                commit(types.GENERATE_RANDOM_HEARD_LIST, list.length);
+                randomHeardList = getters.queue.randomHeardList;
+            }
+            if (randomHeardListPointer == randomHeardList.length - 1) {
+                commit(types.SET_RANDOMLIST_POINTER, 0);
+                nextIndex = randomHeardList[0];
+            }
+            else {
+                nextIndex = randomHeardList[randomHeardListPointer + 1];
+                commit(types.SET_RANDOMLIST_POINTER, randomHeardListPointer + 1);
+            }
```

Comment:
这里其实没有必要用 `if else` ，直接计算可能更好：

```suggestion
            const newPointer = (randomHeardListPointer + 1) % randomHeardList.length;
            commit(types.SET_RANDOMLIST_POINTER, newPointer);
            nextIndex = randomHeardList[randomHeardListPointer];
```